exporter/kafkaexporter/topic/kafka_exporter_ctx.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package topic // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter/topic"
```

Comment:
Could this be moved to `internal/kafka/topic` instead? Would that remove the need to add this to `cmd/checkapi/allowlist.txt`?

Reply:
No, because the intent is for other components to be able to specify the topic via this opaque context key mechanism.

Reply:
> No, because the intent is for other components to be able to specify the topic via this opaque context key mechanism.

Will these components be out of Collector contrib?

I don't think this should be in `exporter/kafkaexporter`, if `internal` is not the right place, then [`pkg`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg) would be the right option. Component modules are explicitly not meant to have arbitrary public API, it should just be config and the `NewFactory` function